src/coreclr/vm/class.h

```diff
@@ -1387,13 +1387,17 @@ class EEClass // DO NOT CREATE A NEW EEClass USING NEW!
     inline void SetChunks (MethodDescChunk* pChunks)
     {
         LIMITED_METHOD_CONTRACT;
-        m_pChunks = pChunks;
+        _ASSERTE(pChunks != NULL);
+        _ASSERTE(m_pChunksBegin == NULL);
+        _ASSERTE(m_pChunksCurrent == NULL);
+        m_pChunksBegin = pChunks;
```

Comment:
I do not think that this is correct. `MethodDescChunk* pChunks` can be a linked list of chunks. so this would have to walk through the linked list to find the end to compute the current chunk.

Is this refactoring related the rest of the changes? I think it would be best to do it separately. There are a lot of subtle details on how it works.

I assume that this change was motivated by `AddChunk` call from dynamicmethod.cpp since it does more than necesary. I would be ok with introducing `AddChunkUnordered` that just adds the new chunk to the start of the linked list and call it from dynamicmethod.cpp. The order for dynamic methods does not matter, so it is easy to reason about. But I would drop the larger refactoring that introduces the second pointer.